### PR TITLE
chore(checkout): CHECKOUT-8562 Update typescript from 4.9.5 to 5.6.2

### PR DESCRIPTION
**Pull request by richapatel1510**
## What?
Upgrade typescript from 4.9.5 to 5.6.2. Upgraded other related dependencies i.e. ts-jest, jest, @types/jest and msw.

## Why?
Update to newer typescript version.

## Testing / Proof

- CI checks
- Manual testing

https://github.com/user-attachments/assets/df9c718b-c406-4a7e-8dc3-e207fee4b0c4

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/core/src/app/checkout/Checkout.spec.tsx [hunk 1]
```diff
@@ -273,7 +273,7 @@ describe('Checkout', () => {
     });
 
     it('calls trackStepViewed when a step is expanded', async () => {
-        jest.useFakeTimers();
+        jest.useFakeTimers({ legacyFakeTimers: true });
```
Comment (richapatel1510) [inline packages/core/src/app/checkout/Checkout.spec.tsx:276]: Need to add this config after upgrading jest `28.0.0`
https://jest-archive-august-2023.netlify.app/docs/28.x/upgrading-to-jest28#jestusefaketimers

File: package.json [hunk 2]
```diff
@@ -147,10 +147,11 @@
     "glob": "^8.0.1",
     "http-server": "^14.1.1",
     "image-webpack-loader": "^8.1.0",
-    "jest": "^26.6.3",
+    "jest": "^29.0.0",
+    "jest-environment-jsdom": "^29.7.0",
```
Comment (richapatel1510) [inline package.json:151]: Need to add this as a separate package for jest version `28.0.0` onwards.
https://jest-archive-august-2023.netlify.app/docs/28.x/upgrading-to-jest28#jsdom

File: packages/adyen-integration/jest.config.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 module.exports = {
     displayName: 'adyen-integration',
     preset: '../../jest.preset.js',
-    globals: {
-        'ts-jest': {
+    transform: {
+        '^.+\\.[tj]sx?$': ['ts-jest', {
```
Comment (richapatel1510) [inline packages/adyen-integration/jest.config.js:5]: Need to move ts-jest config to transform as that configuration in globals is deprecated. 
https://stackoverflow.com/questions/75955311/warn-define-ts-jest-config-under-globals-is-deprecated

File: packages/core/src/app/shipping/ShippingAddressForm.spec.tsx [hunk 4]
```diff
@@ -134,40 +134,4 @@ describe('ShippingAddressForm Component', () => {
             });
         });
     });
-
-    describe('when address is updated', () => {
```
Comment (richapatel1510) [inline packages/core/src/app/shipping/ShippingAddressForm.spec.tsx:138]: removed this, since there were no tests inside this block.

**Review by bc-peng — APPROVED** [review]
LGTM.👏 👏

File: .circleci/config.yml [hunk 5]
```diff
@@ -24,13 +24,23 @@ orbs:
 jobs:
   test-packages:
     <<: *node_executor
+    resource_class: medium+
```
Comment (animesh1987) [inline .circleci/config.yml:27]: Can we remove this back to default and see if tests still pass?

Comment (richapatel1510) [reply .circleci/config.yml]: Tried to remove this [here](https://github.com/bigcommerce/checkout-js/pull/2016/commits/9f5504efbfb506d8145343421a6dbee062e44580), the test-packages does not pass, hence reverting it.

File: packages/core/src/app/customer/Customer.tsx [hunk 6]
```diff
@@ -370,6 +370,8 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps & Ana
 
         try {
             await sendLoginEmail(values);
+        } catch (error) {
+        // Need to write catch block since one test covers the case when `sendLoginEmail` fails
```
Comment (serhii-tkachenko) [inline packages/core/src/app/customer/Customer.tsx:374]: Do you have a ticket to handle the error?

**Review by serhii-tkachenko — APPROVED** [review]
LGTM 👍

**Review by animesh1987 — APPROVED** [review]

File: packages/core/src/app/customer/Customer.tsx (hunk 6 above)
Comment (richapatel1510) [reply packages/core/src/app/customer/Customer.tsx:374]: https://bigcommercecloud.atlassian.net/browse/CHECKOUT-8703

**Pull request merged**